tests/equilibria/test_file.py

```diff
@@ -198,6 +224,32 @@ def test_write_then_read_in_json_format(self):
 
         assert eq2.nz == 151
         assert eq2.nbdry == 72
+        assert eq2.coil_names is None
+        assert eq2.coil_types is None
+
+        # Test with DN-DEMO_eqref_withCoilNames.json (with coil information)
+        eq3 = EQDSKInterface.from_file(self.testfiles[3])
+
+        with open(str(self.testfiles[3])) as data_file:
+            eq3_string = json.load(data_file)
+
+        with mock.patch(OPEN, new_callable=mock.mock_open) as open_mock:
+            eq3.write("some/path.json", file_format="json")
+        written = combine_text_mock_write_calls(open_mock)
+
+        # check the contents of the written mock variable
+        assert_dict_subset(eq3_string, json.loads(written))
```

Comment:
Replace this with the `compare_dicts` function `utilities/tools`.

After looking at it with @je-cook the `eq3_string` and what's returned from `json.loads(written)` should be identical.

Because what is written by the `eq3.write` call, should be identical to the json file on disk already (as it, in theory, should have written it in the past at some point).

Reply:
Is your point that this check is not required?